presto-main/src/main/java/com/facebook/presto/operator/OperatorContext.java

```diff
@@ -634,6 +635,13 @@ private long decrementSpilledReservation(long reservedBytes, long bytesBeingFree
         }
 
         @Override
+        public void close()
+        {
+            // Only products of SpillContext.newLocalSpillContext() should be closed.
+            throw new UnsupportedOperationException(format("%s should be closed directly", getClass()));
```

Comment:
Moved to #8263 (original commit + message fix).
You can't just remove default `close()` implementation because tests use `bytes -> {}` as `SpillContext` mock. We can introduce `TestingSpillContext` instead but I don't see much benefit.